snuba/datasets/configurations/config_loader.py

```diff
@@ -0,0 +1,207 @@
+import collections
+from dataclasses import dataclass, fields
+from typing import (
+    Any,
+    Callable,
+    Mapping,
+    MutableSequence,
+    Optional,
+    Sequence,
+    Type,
+    Union,
+)
+
+from arroyo import Topic as KafkaTopic
+from arroyo.backends.kafka import KafkaProducer
+from arroyo.processing.strategies.dead_letter_queue import (
+    DeadLetterQueuePolicy,
+    ProduceInvalidMessagePolicy,
+)
+
+from snuba.clickhouse.columns import (
+    Array,
+    Column,
+    DateTime,
+    Float,
+    Nested,
+    SchemaModifiers,
+    String,
+    UInt,
+)
+from snuba.datasets.generic_metrics_processor import (
+    GenericDistributionsMetricsProcessor,
+)
+from snuba.datasets.message_filters import KafkaHeaderSelectFilter
+from snuba.datasets.storage import WritableTableStorage
+from snuba.utils.streams.configuration_builder import build_kafka_producer_configuration
+from snuba.utils.streams.topics import Topic
+
+
+def dataclass_from_dict(cls: Type[Any], d: Any) -> Any:
+    # recursively type cast dict to classes
+    # HACK: super hacks
```

Comment:
I know this is a hack so I am not criticizing, just leaving thoughts on how this should work.

This parsing function is assuming that the dataclasses can be arbitrarily combined, which is not true. The structure of our config files will be quite strict, meaning we don't need generic recursive parsers, we can parse each object knowing what it should be. For example we know that a `schema` has a `columns` key and the value is a sequence of `Column` objects. We can parse that directly. If it fails, then the config is bad anyways.